### PR TITLE
Link Zaakdetails object to Zaak.

### PR DESCRIPTION
**Pull request by ThomasMinkeRitense**
### Describe the changes
Link to the related Github issue: https://github.com/generiekzaakafhandelcomponent/gzac-issues/issues/56

**Relevant comments:**
> 

### Breaking changes

- [x] The contribution only contains changes that are not breaking.

### Documentation

- [ ] Release notes have been written for these changes. 

Link to the pull request in the [Valtimo documentation repository](https://github.com/valtimo-platform/valtimo-documentation):
>

New features or changes that have been introduced have been documented.
- [ ] Yes
- [ ] Not applicable

### Tests
Unit tests have been added that cover these changes
- [x] Yes
- [ ] Not applicable

Integration tests have been added that cover these changes
- [x] Yes
- [ ] Not applicable

### Security
The [Secure by Default principle](https://en.wikipedia.org/wiki/Secure_by_default) has been applied to these changes
- [ ] Yes
- [x] Not applicable

Added or changed REST API endpoints have authentication and authorization in place
- [ ] Yes
- [x] Not applicable

Valtimo access control checks have been implemented
- [ ] Yes
- [x] Not applicable

### Dependencies
Newly added dependencies do not introduce known vulnerabilities/CVE's and are in line with the Valtimo license
- [ ] Yes
- [x] Not applicable

### CODE REVIEW

File: zgw/zaakdetails/src/main/kotlin/com/ritense/zaakdetails/domain/ZaakdetailsObject.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import jakarta.persistence.Id
+import jakarta.persistence.Table
+import java.net.URI
+import java.util.UUID
+
+@Entity
+@Table(name = "zaakdetails_object")
+data class ZaakdetailsObject(
+    @Id
+    @Column(name = "id")
```
Comment (ivo-ritense) [inline zgw/zaakdetails/src/main/kotlin/com/ritense/zaakdetails/domain/ZaakdetailsObject.kt]: This has to be the document id based on the way it is used. I would like to see this reflected in the name as well.

File: zgw/zaakdetails/src/test/kotlin/com/ritense/zaakdetails/documentobjectenapisync/DocumentObjectenApiSyncServiceIntTest.kt [hunk 2]
```diff
@@ -69,43 +113,179 @@ class DocumentObjectenApiSyncServiceIntTest : BaseIntegrationTest() {
...
         verify(objectenApiClient, times(1)).createObject(any(), any(), any())
-        verify(objectenApiClient, times(0)).objectUpdate(any(), any(), any())
+        verifyNoMoreInteractions(objectenApiClient)
+        verify(zakenApiClient, times(1)).createZaakObject(any(), any(), any())
     }
 
     @Test
-    fun `should sync modified document to Objecten API`() {
-        documentObjectenApiSyncService.saveSyncConfiguration(
+    fun `should update zaakdetails object and link to the zaak when zaak exists`() {
```
Comment (ivo-ritense) [inline zgw/zaakdetails/src/test/kotlin/com/ritense/zaakdetails/documentobjectenapisync/DocumentObjectenApiSyncServiceIntTest.kt:125]: I think we don't have a test for the case where we already have an zaak details object but it wasn't linked before?

Edit: I guess this situation is covered in the service test.

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/client/ZakenApiClient.kt [hunk 3]
```diff
@@ -476,6 +486,58 @@ class ZakenApiClient(
...
+            }
+            .headers(this::defaultHeaders)
+            .contentType(MediaType.APPLICATION_JSON)
+            .body(request)
+            .retrieve()
+            .body<ZaakObject>()!!
+
+        result.objectUrl = sanitizeUriHost(result.objectUrl)
+
+        //TODO: outbox event
```
Comment (ivo-ritense) [inline zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/client/ZakenApiClient.kt]: TODO?

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/domain/ZaakObjectRequest.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.zakenapi.domain
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import java.net.URI
+import java.util.*
```
Comment (marijnritense) [inline zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/domain/ZaakObjectRequest.kt]: No * imports

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/domain/ZaakObject.kt [hunk 5]
```diff
@@ -26,7 +26,7 @@ data class ZaakObject(
     @JsonProperty("zaak")
     val zaakUrl: URI,
     @JsonProperty("object")
-    val objectUrl: URI,
+    var objectUrl: URI,
```
Comment (marijnritense) [inline zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/domain/ZaakObject.kt]: Can you change this back to val, and use .copy instead?

File: zgw/zaakdetails/src/main/kotlin/com/ritense/zaakdetails/service/ZaakdetailsObjectService.kt [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.zaakdetails.service
+
+import com.ritense.zaakdetails.domain.ZaakdetailsObject
+import com.ritense.zaakdetails.repository.ZaakdetailsObjectRepository
+import java.util.*
```
Comment (marijnritense) [inline zgw/zaakdetails/src/main/kotlin/com/ritense/zaakdetails/service/ZaakdetailsObjectService.kt]: No * imports

**Review by marijnritense — APPROVED** [review]

**Pull request merged**